File: Prefix Sum/1991. Find the Middle Index in Array.cpp
class Solution {
public:
    int findMiddleIndex(vector<int>& nums) {
        int m,startsum=0,ttsum=0;
        ttsum=accumulate(nums.begin(),nums.end(),0);
        
        
        for(int i=0;i<nums.size();i++)
        {
           m=i;
            ttsum-=nums[i];
            if(ttsum==startsum)
            {
                return m;
            }
            else
            {
                startsum+=nums[i];
            }
        }
        if(m==nums.size()-1)
        {
            return -1;
        }
        return m;
        
    }
};

File: Prefix Sum/1991. Find the Middle Index in Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMiddleIndex = function(nums) {
    
    var tsum=0,ssum=0,m;
    for(var i=0;i<nums.length;i++)
        {
            tsum+=nums[i];
        }
    for(var i=0;i<nums.length;i++)
        {
            m=i;
            tsum-=nums[i];
            if(tsum==ssum)
                {
                    return m;
                }
            else
                {
                    ssum+=nums[i];
                }
        }
    if(m==nums.length-1)
        {
            return -1;
        }
    return m;
};
